### PR TITLE
API to get type information of a given expression

### PR DESCRIPTION
**Pull request by sajeerAhamed15**
**Request**
```
{
   "documentIdentifier":{
      "uri":<file_uri>
   },
   "startPosition":{
      "line":1,
      "offset":1
   },
   "endPosition":{
      "line":1,
      "offset":1
   }
}
```

**Response**
```
{ TypeSymbol }
```

Fixes [#29188](https://github.com/ballerina-platform/ballerina-lang/issues/29188)

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Comment by anupama-pathirage** [comment]
@sajeerAhamed15 The commit messages and PR heading is not according to the best practices. Please correct them.

A properly formed Git commit subject line should always be able to complete the following sentence:

If applied, this commit will **_your subject line here_**

Refer [1] for more details.

[1] https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution

**Comment by sajeerAhamed15** [comment]
> @sajeerAhamed15 The commit messages and PR heading is not according to the best practices. Please correct them.
> 
> A properly formed Git commit subject line should always be able to complete the following sentence:
> 
> If applied, this commit will **_your subject line here_**
> 
> Refer [1] for more details.
> 
> [1] https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution

ack. I'll squash the commits as well

File: language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/ExpressionTypeRequest.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```
Comment (kalaiyarasiganeshalingam) [inline language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/extensions/ballerina/symbol/ExpressionTypeRequest.java:2]:
```suggestion
 * Copyright (c) 2021, WSO2 Inc. (http://wso2.com) All Rights Reserved.
```

**Comment by sajeerAhamed15** [comment]
Closing this PR because of Duplicate PR [#29920](https://github.com/ballerina-platform/ballerina-lang/pull/29920)

**Pull request closed without merging**